### PR TITLE
#174: Upgrade testcontainers dependency (and other dependencies).

### LINKED ISSUE #174: Upgrade testcontainers dependency
[testcontainers-java](https://github.com/testcontainers/testcontainers-java) version [1.16.1](https://github.com/testcontainers/testcontainers-java/releases/tag/1.16.1) upgrades junit to 4.13.2 (fixing a vulnerability) and fixes an [issue running tests on macOS](https://github.com/testcontainers/testcontainers-java/issues/4384).

That's why we should upgrade to the latest version.

Also upgrade JUnit Jupiter to the latest release 5.8.x

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #174.

### CODE REVIEW

File: doc/changes/changes_5.1.2.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Test containers for Exasol on Docker 5.1.2, released 2021-11-25
```
Comment (kaklakariada) [inline doc/changes/changes_5.1.2.md:1]:
```suggestion
# Test containers for Exasol on Docker 5.1.2, released 2021-11-26
```

File: pom.xml [hunk 2]
```diff
@@ -384,20 +377,20 @@
...
+            </goals>
+            <configuration>
+                <rules>
+                    <requireMavenVersion>
+                        <version>3.6.3</version>
+                    </requireMavenVersion>
+                </rules>
+            </configuration>
+        </execution>
+    </executions>
```
Comment (kaklakariada) [inline pom.xml:393]: Format this file

File: src/main/java/com/exasol/containers/ContainerFileOperations.java [hunk 3]
```diff
@@ -15,6 +15,11 @@
 
     private final ExasolContainer<? extends ExasolContainer<?>> container;
 
+    /**
+     * Creates a new instance of the {@link ContainerFileOperations}.
```
Comment (kaklakariada) [inline src/main/java/com/exasol/containers/ContainerFileOperations.java:19]:
```suggestion
     * Create a new instance of the {@link ContainerFileOperations}.
```
❓

File: src/main/java/com/exasol/containers/ContainerFileOperations.java [hunk 4]
```diff
@@ -15,6 +15,11 @@
...
+     * 
+     * @param container Container reference required executing commands inside the container.
```
Comment (kaklakariada) [inline src/main/java/com/exasol/containers/ContainerFileOperations.java:21]:
```suggestion
     * @param container Container reference required for executing commands inside the container.
```
❓

File: src/main/java/com/exasol/containers/ExasolContainerConstants.java [hunk 5]
```diff
@@ -6,29 +6,64 @@
  * Constants for the Exasol Docker containers (like image ID and version).
  */
 public final class ExasolContainerConstants {
+    /** Version of the Exasol Docker image */
     public static final String EXASOL_DOCKER_IMAGE_VERSION = "7.1.1";
```
Comment (kaklakariada) [inline src/main/java/com/exasol/containers/ExasolContainerConstants.java:10]:
```suggestion
    public static final String EXASOL_DOCKER_IMAGE_VERSION = "7.1.2";
```

File: src/main/java/com/exasol/support/SupportInformationRetriever.java [hunk 6]
```diff
@@ -18,7 +18,9 @@
  * Manages getting support information (like cluster logs, configuration settings and core-dumps) from the database.
  */
 public class SupportInformationRetriever {
+    /** Target directory where support information packages will be stored */
```
Comment (kaklakariada) [inline src/main/java/com/exasol/support/SupportInformationRetriever.java:21]:
```suggestion
    /** Name of the property that allows overriding the target directory where support information packages will be stored */
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]
Some minor changes requested

File: pom.xml (hunk 2 above)
Comment (redcatbear) [reply pom.xml]: Done.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**